### PR TITLE
docs(b and c components): consistent api description refs

### PR DESCRIPTION
**Pull request by DitwanP**
**Related Issue:** #7071

## Summary
Updates doc consistency across b and c-named components defined in the above issue for props, events, methods, and css vars, including:

- `button`
- `block`
- `block-section`
- `color-picker-hex-input`
- `color-picker`
- `combobox`

cc @geospatialem

### CODE REVIEW

File: packages/calcite-components/src/components/block-section/block-section.tsx [hunk 1]
```diff
@@ -65,7 +65,7 @@ export class BlockSection implements LocalizedComponent, T9nComponent, LoadableC
   @Prop() text: string;
 
   /**
-   * Specifies the component's toggle display -
+   * Specifies how the component's toggle is displayed -
```
Comment (geospatialem) [inline packages/calcite-components/src/components/block-section/block-section.tsx:68]: 😍

File: packages/calcite-components/src/components/block/block.tsx [hunk 2]
```diff
@@ -251,7 +251,7 @@ export class Block
   /**
    * Emits when the component's header is clicked.
    *
-   * @deprecated use `openClose` events: `calciteBlock[Before]Open` and `calciteBlock[Before]Close` instead.
+   * @deprecated use `openClose` events: `calciteBlockBeforeOpen` and `calciteBlockBeforeClose` instead.
```
Comment (geospatialem) [inline packages/calcite-components/src/components/block/block.tsx:254]: nitpick: Capitalize Use.

Comment (geospatialem) [reply packages/calcite-components/src/components/block/block.tsx:254]: Would this also apply to four events - `calciteBlockBeforeOpen`, `calciteBlockOpen`, `calciteBlockBeforeClose`, and `calciteBlockClose`?

File: packages/calcite-components/src/components/color-picker/color-picker.tsx [hunk 3]
```diff
@@ -116,7 +118,7 @@ export class ColorPicker
     }
   }
 
-  /** When true, hides the RGB/HSV channel inputs */
+  /** When `true`, hides the RGB/HSV channel inputs */
```
Comment (geospatialem) [inline packages/calcite-components/src/components/color-picker/color-picker.tsx:121]: Awesome! It looks like we could add in the period at the end of this line as well?

File: packages/calcite-components/src/components/color-picker/color-picker.tsx [hunk 4]
```diff
@@ -161,7 +163,7 @@ export class ColorPicker
    */
   @Prop({ reflect: true }) hideChannels = false;
 
-  /** When true, hides the hex input */
+  /** When `true`, hides the hex input */
```
Comment (geospatialem) [inline packages/calcite-components/src/components/color-picker/color-picker.tsx:166]: It looks like another period could be added at the end this line.

File: packages/calcite-components/src/components/color-picker/color-picker.tsx [hunk 5]
```diff
@@ -178,7 +180,7 @@ export class ColorPicker
    */
   @Prop({ reflect: true }) hideSaved = false;
 
-  /** When true, hides the saved colors section */
+  /** When `true`, hides the saved colors section */
```
Comment (geospatialem) [inline packages/calcite-components/src/components/color-picker/color-picker.tsx:183]: And one moar period could be added at the end this line.

File: packages/calcite-components/src/components/combobox/combobox.tsx [hunk 6]
```diff
@@ -219,10 +219,10 @@ export class Combobox
...
   /**
    * Specifies the selection mode:
-   * - "multiple" allows any number of selected items (default),
-   * - "single" allows only one selection,
-   * - "single-persist" allow and require one open item,
-   * - "ancestors" is like multiple, but shows ancestors of selected items as selected, with only deepest children shown in chips.
+   * - `"multiple"` allows any number of selected items (default),
+   * - `"single"` allows only one selection,
+   * - `"single-persist"` allow and require one open item,
+   * - `"ancestors"` is like multiple, but shows ancestors of selected items as selected, with only deepest children shown in chips.
```
Comment (geospatialem) [inline packages/calcite-components/src/components/combobox/combobox.tsx:225]: 🧹

**Review by geospatialem — APPROVED** [review]
Nice! ✨ 🧹 ✨ 

Have a few considerations for your review below.

https://github.com/Esri/calcite-design-system/blob/f01803193b5f381a2dca507c5468fca62fcfbcfc/packages/calcite-components/src/components/button/button.tsx#L153

For consistency, we should rename any named components, something like:
> Specifies the default behavior of the ~button~ component. 

https://github.com/Esri/calcite-design-system/blob/f01803193b5f381a2dca507c5468fca62fcfbcfc/packages/calcite-components/src/components/chip-group/chip-group.tsx#L52-L56

Could we add some context to the selection mode types, similar to [list](https://github.com/Esri/calcite-design-system/blob/main/packages/calcite-components/src/components/list/list.tsx#L191-L197)?


https://github.com/Esri/calcite-design-system/blob/f01803193b5f381a2dca507c5468fca62fcfbcfc/packages/calcite-components/src/components/checkbox/checkbox.tsx#L87-L88

WDYT about adding in context to the `name` prop, similar to [combobox](https://github.com/Esri/calcite-design-system/blob/main/packages/calcite-components/src/components/combobox/combobox.tsx#L188-L196)?

File: packages/calcite-components/src/components/block/block.tsx (hunk 2 above)
Comment (DitwanP) [reply packages/calcite-components/src/components/block/block.tsx:254]: Yeah I think so, will mention them.

File: packages/calcite-components/src/components/chip-group/chip-group.tsx [hunk 7]
```diff
@@ -49,7 +49,13 @@ export class ChipGroup implements InteractiveComponent {
...
   @Prop({ reflect: true }) scale: Scale = "m";
 
-  /** Specifies the selection mode of the component. */
+  /**
+   * Specifies the selection mode of the component:
+   * - `"multiple"` allows any number of selections,
+   * - `"single"` allows only one selection,
+   * - `"single-persist"` allow and require one selected item,
+   * - `"none"` does not allow any selections.
+   */
```
Comment (geospatialem) [inline packages/calcite-components/src/components/chip-group/chip-group.tsx:58]: Small nitpick: _Believe_ in this case both the `:` and `-` / `,` and `-` would be shown together. Maybe remove the `-` throughout so only one character is displayed in the doc?

File: packages/calcite-components/src/components/combobox/combobox.tsx [hunk 8]
```diff
@@ -219,10 +219,10 @@ export class Combobox
...
   /**
    * Specifies the selection mode:
-   * - "multiple" allows any number of selected items (default),
-   * - "single" allows only one selection,
-   * - "single-persist" allow and require one open item,
-   * - "ancestors" is like multiple, but shows ancestors of selected items as selected, with only deepest children shown in chips.
+   * - `"multiple"` allows any number of selected items,
+   * - `"single"` allows only one selection,
+   * - `"single-persist"` allow and require one open item,
+   * - `"ancestors"` is like multiple, but shows ancestors of selected items as selected, with only deepest children shown in chips.
```
Comment (geospatialem) [inline packages/calcite-components/src/components/combobox/combobox.tsx:225]: Similar to the other instance _believe_ in this case both the `:` and `-` / `,` and `-` would be shown together. Maybe remove the `-` throughout so only one character is displayed in the doc? For instance, a screenshot of the [combobox page](https://developers.arcgis.com/calcite-design-system/components/combobox/):

image

**Review by geospatialem — APPROVED** [review]
Thanks for working through these! 🙌🏻 Two small considerations from some duplicative styling where the prop values are summarized.

**Pull request merged**